models/experimental/functional_stable_diffusion3_5/ttnn/common.py

```diff
@@ -0,0 +1,106 @@
+            )
+        self.groups = groups
+        self.dilation = dilation
+        self.use_shallow_conv_variant = (use_shallow_conv_variant,)
+
+    def __call__(self, device, input_tensor):
+        if input_tensor.shape[3] == 1024 or input_tensor.shape[3] == 384:
+            input_tensor = ttnn.to_memory_config(input_tensor, memory_config=ttnn.DRAM_MEMORY_CONFIG)
+        else:
+            input_tensor = ttnn.from_device(input_tensor)
+        conv_config = ttnn.Conv2dConfig(
+            dtype=self.dtype,
+            weights_dtype=ttnn.bfloat16,
+            activation=self.activation,
+            shard_layout=self.shard_layout,
+            input_channels_alignment=(
+                16
+                if self.use_shallow_conv_variant or (input_tensor.shape[3] == 16 and input_tensor.shape[1] == 115)
+                else 32
+            ),
+            # reshard_if_not_optimal=self.reshard,
+            deallocate_activation=self.deallocate,
+            reallocate_halo_output=True,
+            enable_act_double_buffer=False,
+            enable_split_reader=False,
+            enable_subblock_padding=False,
+            output_layout=ttnn.TILE_LAYOUT,
+        )
+        compute_config = ttnn.init_device_compute_kernel_config(
+            device.arch(),
+            math_fidelity=ttnn.MathFidelity.LoFi,
+            fp32_dest_acc_en=False,
+            packer_l1_acc=False,
+            math_approx_mode=True,
+        )
+        if self.act_block_h is not None:
+            conv_config.act_block_h_override = self.act_block_h
+        [output_tensor, [_out_height, _out_width]] = ttnn.conv2d(
+            input_tensor=input_tensor,
+            weight_tensor=self.weights,
+            bias_tensor=self.bias,
+            in_channels=input_tensor.shape[3],
+            out_channels=self.out_channels,
+            device=device,
+            kernel_size=self.kernel_size,
+            stride=(self.conv_params[0], self.conv_params[1]),
+            padding=(self.conv_params[2], self.conv_params[3]),
+            batch_size=input_tensor.shape[0],
+            input_height=input_tensor.shape[1],
+            input_width=input_tensor.shape[2],
+            conv_config=conv_config,
+            debug=False,
+            groups=self.groups,
+            return_output_dim=True,
+            return_weights_and_bias=False,
+            compute_config=compute_config,
+            dilation=(self.dilation, self.dilation),
+        )
+        output_tensor = ttnn.sharded_to_interleaved(output_tensor, ttnn.L1_MEMORY_CONFIG)
+        output_tensor = ttnn.to_layout(output_tensor, layout=ttnn.ROW_MAJOR_LAYOUT)
+        output_tensor = ttnn.reshape(
+            output_tensor, (input_tensor.shape[0], _out_height, _out_width, output_tensor.shape[3])
+        )
+        output_tensor = ttnn.to_layout(output_tensor, layout=ttnn.TILE_LAYOUT)
+        del _out_height, _out_width
```

Comment:
Is this necessary? These are tiny objects, and will be automatically cleaned up at the end of this scope.

Reply:
I will remove it.